### PR TITLE
Remove py kwargs

### PR DESCRIPTION
**Pull request by Sharpz7**
Closes https://github.com/G-Research/armada/issues/1194

### CODE REVIEW

**Review by kannon92 — CHANGES_REQUESTED** [review]
Can you update the unit tests with the new calls to this functions?

File: client/python/armada_client/client.py [hunk 1]
```diff
@@ -146,29 +146,67 @@ def reprioritize_jobs(
         response = self.submit_stub.ReprioritizeJobs(request)
         return response
 
-    def create_queue(self, name: str, **queue_params) -> empty_pb2.Empty:
+    def create_queue(
+        self,
+        name: str,
+        priority_factor: Optional[int],
```
Comment (kannon92) [inline client/python/armada_client/client.py:152]: This is actually a double in the API definition.  I think double->float is fine?  
```suggestion
        priority_factor: Optional[float],
```

```
    double priority = 1;
```

File: client/python/armada_client/client.py [hunk 2]
```diff
@@ -146,29 +146,67 @@ def reprioritize_jobs(
...
+        user_owners: Optional[List[str]] = None,
+        group_owners: Optional[List[str]] = None,
+        resource_limits: Optional[Dict[str, float]] = None,
+        permissions: Optional[List[submit_pb2.Queue.Permissions]] = None,
```
Comment (kannon92) [inline client/python/armada_client/client.py:156]: Should we go a little farther with Permissions?  Have this also take pythonic types as opposed to the proto representations?

Comment (Sharpz7) [reply client/python/armada_client/client.py:156]: Would we want this done in a seperate PR, and leave this as is for now? Or do that in this one?

Comment (kannon92) [reply client/python/armada_client/client.py:156]: I'd put it in this one.

File: client/python/armada_client/client.py [hunk 3]
```diff
@@ -235,16 +273,41 @@ def create_job_request(
         )
 
     def create_job_request_item(
-        self, pod_spec: core_v1.PodSpec, priority: int = 1, **job_item_params
+        self,
+        pod_spec: core_v1.PodSpec,
+        priority: int = 1,
```
Comment (kannon92) [inline client/python/armada_client/client.py:278]:
```suggestion
        priority: float = 1,
```

File: client/python/armada_client/client.py [hunk 4]
```diff
@@ -146,29 +146,67 @@ def reprioritize_jobs(
...
+            permissions=permissions,
+        )
         response = self.submit_stub.CreateQueue(request)
         return response
 
-    def update_queue(self, name: str, **queue_params) -> None:
+    def update_queue(
+        self,
+        name: str,
+        priority_factor: Optional[int],
```
Comment (kannon92) [inline client/python/armada_client/client.py:184]:
```suggestion
        priority_factor: Optional[float],
```

File: client/python/armada_client/client.py (hunk 2 above)
Comment (Sharpz7) [reply client/python/armada_client/client.py:156]: Cool! Will work on this shortly :))

File: client/python/armada_client/client.py [hunk 5]
```diff
@@ -235,16 +273,41 @@ def create_job_request(
         )
 
     def create_job_request_item(
-        self, pod_spec: core_v1.PodSpec, priority: int = 1, **job_item_params
+        self,
+        pod_spec: core_v1.PodSpec,
+        priority: float = 1,
```
Comment (kannon92) [inline client/python/armada_client/client.py:278]:
```suggestion
        priority: float = 1.0,
```

nit pick.  Python probably doesn't even care about this.

File: docs/python_armada_client.md [hunk 6]
```diff
@@ -416,3 +460,61 @@ Definition can be found at:
...
+
+    
+    * **kind** (*str*) – 
+
+
+    * **name** (*str*) – 
+
+
+
+#### kind(_: st_ )
```
Comment (Sharpz7) [inline docs/python_armada_client.md:506]: @jayofdoom any suggestions on how to try and remove the duplicate documentaition of these types?

**Review by kannon92 — APPROVED** [review]

File: docs/python_armada_client.md (hunk 6 above)
Comment (jayofdoom) [reply docs/python_armada_client.md:506]: Lets merge this as-is, and tackle this separately. I have a few ideas but no need to hold up this PR on it.

**Review by jayofdoom — APPROVED** [review]

**Review by kannon92 — APPROVED** [review]

**Pull request merged**